File: backend/app/schemas/repo.py
"""Repository Pydantic schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.schemas.common import TagRef


class RepoSummary(BaseModel):
    id: str
    full_name: str
    description: Optional[str] = None
    url: str
    stars: int
    forks: int
    language: Optional[str] = None
    topics: list[str] = Field(default_factory=list)
    pushed_at: Optional[datetime] = None
    crawled_at: datetime
    last_crawled_at: datetime
    tags: list[TagRef] = Field(default_factory=list)
    summary_cn: Optional[str] = None
    summary_en: Optional[str] = None

    model_config = {"from_attributes": True}


class RepoDetail(RepoSummary):
    """In V1, RepoDetail is identical to RepoSummary."""
    pass


class RepoTagUpdate(BaseModel):
    tag_ids: list[str]


class RepoTagResponse(BaseModel):
    repo_id: str
    tags: list[TagRef]

File: frontend/src/types/repo.ts
import type { TagRef } from './tag';

// ---- Repo Summary (list item) ----
export interface RepoSummary {
  id: string;
  full_name: string;
  description: string | null;
  url: string;
  stars: number;
  forks: number;
  language: string | null;
  topics: string[];
  pushed_at: string | null;
  crawled_at: string;
  last_crawled_at: string;
  tags: TagRef[];
  summary_cn: string | null;
  summary_en: string | null;
}

// ---- Repo Detail (same as summary in V1) ----
export type RepoDetail = RepoSummary;

// ---- Repo Query Params ----
export interface RepoQueryParams {
  page?: number;
  page_size?: number;
  language?: string[];
  topic?: string[];
  stars_min?: number;
  stars_max?: number;
  pushed_after?: string;
  pushed_before?: string;
  tag_id?: string[];
  q?: string;
  sort?: string;
}
